### PR TITLE
Fixing bugs and adding features toward GA

### LINKED ISSUE #103: Add label for each Parameter to localize
Label should add in English to translate. Currently, `Subscription` and `Environment` don't have it.
image
image

### PR DESCRIPTION
**Pull request by tsubasaxZZZ**
Close #103 #121

### CODE REVIEW

File: build/template_kql/azuresiterecovery/azuresiterecovery_resources_details.kql [hunk 1]
```diff
@@ -1,3 +1,5 @@
 ${extend_resource}
-| project Type, SkuName, Kind, SubscriptionId, Name, State, Location, ResourceGroup, Zones, AvailabilityZone, AvailabilitySet, Capacity, FaultDomain, OSDisk, ReadReplica, AutomaticFailover, MultipleWriteLocations, StorageAutogrow, ReplicationRole, Backup = tolower(Backup), Environment
-| where Type == 'microsoft.recoveryservices/vaults'
\ No newline at end of file
+| summarize asrRrotectedTotalItemCounts = count(asrId != ""), asrUnprotectedItemCounts = countif(protectionStateDescription != "Protected"), asrRelicationUnHealthyItemCounts = countif(replicationHealth != "Normal"), asrFailoverUnHealthyItemCounts = countif(failoverHealth != "Normal"), asrIsReplicationAgentUpdateRequiredItemCounts = countif(isReplicationAgentUpdateRequired == true) by Type, SkuName, Kind, SubscriptionId, Name, State, Location, ResourceGroup, Zones, AvailabilityZone, AvailabilitySet, Capacity, FaultDomain, OSDisk, ReadReplica, AutomaticFailover, MultipleWriteLocations, StorageAutogrow, ReplicationRole, Backup, Environment
+| where asrRrotectedTotalItemCounts != 0
```
Comment (infosatheesh2020) [inline build/template_kql/azuresiterecovery/azuresiterecovery_resources_details.kql:3]: Typo here: asrRrotectedTotalItemCounts

File: build/template_kql/common/extend_resource.kql [hunk 2]
```diff
@@ -398,10 +398,25 @@ resources
     properties
 | join kind = leftouter (
     recoveryservicesresources
-    | where name startswith "VM;"
-    | project Name = tolower(tostring(properties.dataSourceInfo.resourceID)), Backup = tostring(properties.protectionStatus)
+    | extend vmId = case(
+        name startswith "VM;", tolower(tostring(properties.dataSourceInfo.resourceID)),
```
Comment (infosatheesh2020) [inline build/template_kql/common/extend_resource.kql:402]: Check possibility of replacing below condition
name startswith "VM;" to  properties.backupManagementType == "AzureIaasVM" in app places where we want to pick VM protection information

**Comment by tsubasaxZZZ** [comment]
@infosatheesh2020 I've tried fix the summary of PIP. Please comment for any feedback.
image

File: build/templates/networking.tpl.json [hunk 3]
```diff
@@ -611,11 +643,15 @@
                     },
                     {
                       "columnId": "NotSucceeded",
-                      "label": "IP Not Succeeded"
+                      "label": "Failed"
```
Comment (tsubasaxZZZ) [inline build/templates/networking.tpl.json:646]: Changed expression to this. The current translated Japanese on Portal is a bit strange, so I'll try changing and then check it after deployment.

File: build/template_kql/common/extend_resource.kql (hunk 2 above)
Comment (tsubasaxZZZ) [reply build/template_kql/common/extend_resource.kql:402]: According to the API reference, this value will be fixed, so we can use it.
https://learn.microsoft.com/en-us/azure/templates/microsoft.recoveryservices/vaults/backuppolicies?pivots=deployment-language-bicep#protectionpolicy

File: build/template_kql/azuresiterecovery/azuresiterecovery_resources_details.kql (hunk 1 above)
Comment (tsubasaxZZZ) [reply build/template_kql/azuresiterecovery/azuresiterecovery_resources_details.kql]: Fixed

File: build/template_kql/common/extend_resource.kql (hunk 2 above)
Comment (tsubasaxZZZ) [reply build/template_kql/common/extend_resource.kql]: Fixed.

**Comment by infosatheesh2020** [comment]
> @infosatheesh2020 I've tried fix the summary of PIP. Please comment for any feedback. What has changed: [tsubasaxZZZ@d6c6ca3](https://github.com/tsubasaxZZZ/reliability-workbook/commit/d6c6ca39cd0e70b1ab4f76598577bffa1f983065)
> 
> Underbar means number of right part.
> 
> image

@tsubasaxZZZ Didn't we also agreed to highlight single zone IP addresses?

**Comment by tsubasaxZZZ** [comment]
@infosatheesh2020 
> @tsubasaxZZZ just for my understanding: Single zone IP = No zone redundant - No zone IP?

No. Single zone IP = No zone redundant.
Public IP's zone options is Zone redundant or not(choose from 1/2/3 or redundant).
image

I could be misleading when we talk. "Single zone" could be good expression better than now.

File: build/templates/storage.tpl.json [hunk 4]
```diff
@@ -234,17 +263,332 @@
...
+                    }
+                  ]
+                }
+              },
+              {
+                "columnMatch": "SubscriptionName",
+                "formatter": 5
+              },
+              {
+                "columnMatch": "SecondaryLocation",
```
Comment (infosatheesh2020) [inline build/templates/storage.tpl.json:399]: @tsubasaxZZZ I don't see these fields (incl secondary status, access tier) in storage KQL. Am I missing something?

File: build/template_kql/database/database_sqldb_resources_details_summary.kql [hunk 5]
```diff
@@ -0,0 +1,4 @@
+${extend_resource}
+| project Type, SkuName, Kind, SubscriptionId, Name, State, Location, ResourceGroup, Zones, AvailabilityZone, AvailabilitySet, Capacity, FaultDomain, OSDisk, ReadReplica, AutomaticFailover, MultipleWriteLocations, StorageAutogrow, ReplicationRole, Backup = tolower(Backup), Environment
+| where Type == 'microsoft.sql/servers/databases' and SkuName !contains "datawarehouse" and Kind !contains "system"
+| summarize ResourceTotal=count(Name), NotRunning=dcountif(Name, State !in~ ("online", "paused")), NotRecommendedSKU=dcountif(Name, SkuName in~ ("standard", "basic")), NoAVZone=dcountif(Name, AvailabilityZone == "not configured"), NoGRSBackup=dcountif(Name, OSDisk !~ "grs")
```
Comment (infosatheesh2020) [inline build/template_kql/database/database_sqldb_resources_details_summary.kql:4]: Is there a way to filter further on the NotRecommendedSKU condition
- Basic to be there.
- Standard S0,S1,S2 to be there. We do allow S3 and above for prod: https://learn.microsoft.com/en-us/azure/azure-sql/database/service-tiers-dtu?view=azuresql#compare-service-tiers

Check the important section in this document. 

This is not just for summary, maybe we need to update for details section as well.

Also UI label for this field can be changed from "Not Production SKU" to "Non-production SKU"

File: build/templates/databases.tpl.json [hunk 6]
```diff
@@ -224,6 +227,76 @@
...
+                              "columnId": "ResourceTotal",
+                              "label": "Resource Total"
+                            },
+                            {
+                              "columnId": "NotRunning",
+                              "label": "Failed Status"
+                            },
+                            {
+                              "columnId": "NotRecommendedSKU",
+                              "label": "Not production SKU"
```
Comment (infosatheesh2020) [inline build/templates/databases.tpl.json:285]: Change "Not production SKU" -> "Non-production SKU"

File: build/templates/storage.tpl.json (hunk 4 above)
Comment (tsubasaxZZZ) [reply build/templates/storage.tpl.json:399]: Maybe these settings just stay in place. I'll remove them.

File: build/template_kql/database/database_sqldb_resources_details_summary.kql (hunk 5 above)
Comment (tsubasaxZZZ) [reply build/template_kql/database/database_sqldb_resources_details_summary.kql:4]: I've tried that.
image

**Pull request merged**